packages/playground/data-liberation/docs-importer-test.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+require_once __DIR__ . '/bootstrap.php';
+
+$reader = new WP_Serialized_Pages_Reader(__DIR__ . '/../../docs/site/docs');
```

Comment:
Just want to make sure I'm following correctly. 

Is the idea here that any project would implement their own instance of this `WP_Serialized_Pages_Reader` in order to customize how front-matter should be read from that specific project? 

For example, in my WPGraphQL Docs, I have frontmatter like so: 

```
title: Contributing
uri: `/docs/contributing`
```

So, following this example, I could map my specific front-matter in the .md files I'm importing to whatever field(s) I want them to map to when being imported as a WordPress post? 

Or is this something that's always running when front-matter is detected in .md and there would be a different mechanism (add_filter, for example) to custom map front-matter keys/values to WordPress (wxr) keys/values?

Reply:
> So, following this example, I could map my specific front-matter in the .md files I'm importing to whatever field(s) I want them to map to when being imported as a WordPress post?

Exactly! I think every project will have to handle its own frontmatter. I'm not aware of any unified schema for frontmatter metadata and I've seen a few different variations.

Reply:
Although I like `add_filter()` maybe even more – it makes extensibility easier

Reply:
Ya, I was thinking there could be some documented defaults.

i.e. 

| Frontmatter | WXR         |   |   |   |
|-------------|-------------|---|---|---|
| title       | post_title  |   |   |   |
| date        | post_date   |   |   |   |
| status      | post_status |   |   |   |
| ...         | ...         |   |   |   |

Then if folks use those documented defaults in frontmatter already, things will "just work", but if they need to customize the mappings they could do so via a custom php snippet added in "steps" or a custom plugin loaded or whatever 🤔 

Here's a (pseudo) example: 

```php
add_filter( 'wp_playground_map_front_matter', function( $wxr, $unfiltered_frontmatter ) { 
  // do some logic based on the frontmatter key and map it to wxr
  if ( isset( $unfiltered_frontmatter['something'] ) {
    $wxr['wp:some_meta_key'] => $unfiltered_frontmatter['something'];
  }
  return $wxr;
} );
```

Reply:
I like that! I'd just keep the entire WXR pipeline completely separate and treat Markdown -> WordPress independently. In this scenario, we'd map the frontmatter keys directly to `wp_insert_post` keys.